instructor/common.py

```diff
@@ -18,30 +18,41 @@
 
 
 @login_required
-def assignments(request):
+def assignments(request, error=''):
     assignments = models.Assignment.objects.filter(course__owner=request.user)
+    return render_to_response(
+        'instructor/assignments.html',
+        {
+            'error': json.dumps(error),
```

Comment:
You have `dumps` here, which takes a python object and turns it to a string, but the urlresolver will only ever give you strings.

Also, accepting json as a url fragment instead of as a post variable or query param seems quite odd. Also, `/` is valid for json contents. Perhaps I'm missing some important context here.